server/src/packages/landuse-form/landuse-actions/landuse-actions.controller.ts

```diff
@@ -29,16 +29,35 @@ export class LanduseActionsController {
   @Patch('/:id')
   async update(@Body() body, @Param('id') id) {
     const allowedAttrs = pick(body, LANDUSE_ACTION_ATTRS);
+    const dateOfPreviousApproval = await this.setDate(body.dcp_dateofpreviousapproval);
+    const lapseDateOfPreviousApproval = await this.setDate(body.dcp_lapsedateofpreviousapproval);
+    const recordationDate = await this.setDate(body.dcp_recordationdate);
 
     await this.crmService.update('dcp_landuseactions', id, {
       ...allowedAttrs,
 
       ...(body.chosen_zoning_resolution_id ? { 'dcp_zoningresolutionsectionactionispursuantto@odata.bind': `/dcp_zoningresolutions(${body.chosen_zoning_resolution_id})` } : {}),
+      dcp_dateofpreviousapproval: dateOfPreviousApproval,
+      dcp_lapsedateofpreviousapproval: lapseDateOfPreviousApproval,
+      dcp_recordationdate: recordationDate,
     });
 
     return {
       dcp_landuseactionid: id,
       allowedAttrs,
     };
   }
+
+  // when a user selects a date from the frontend date-picker, this value
+  // is sent to backend as a datestring in an array, so we need to grab
+  // the first item in that array. When the user does not use the date-picker,
+  // and instead the date value comes from CRM (still exists in the body), the
+  // date is just formatted as an isolated datestring and does not need to be reformatted.
+  async setDate(bodyDateField) {
```

Comment:
It looks like maybe `pickDate` would be a better name? This function isn't setting anything, but retrieving a date field from an array, if available. It looks like it serves a similar purpose to the `pick()` function, but customized to dates. 

Also this doesn't need to be async. I don't see any asynchronous functions within setDate.

Reply:
Okay sounds good